File: src/main/java/com/ata1635/cardgame/core/model/Card.java
package com.ata1635.cardgame.core.model;

public class Card {
    private final String value;
    private final String type;

    public Card(String value, String type) {
        this.value = value;
        this.type = type;
    }

    public String toString() {
        return value + "-" + type;
    }

    public int getValue() {
        switch (value) {
            case "A": return 11;
            case "J":
            case "Q":
            case "K": return 10;
            default: return Integer.parseInt(value); //2-10
        }
    }

    public boolean isAce() {
        return "A".equals(value);
    }

    public String getImagePath() {
        return "/cards/" + toString() + ".png";
    }
}

File: src/js/core/model/Card.js
export class Card {
    constructor(value, type) {
        this.value = value;
        this.type = type;
    }

    toString() {
        return `${this.value}-${this.type}`;
    }

    getValue() {
        switch (this.value) {
            case "A": return 11;
            case "J":
            case "Q":
            case "K": return 10;
            default: return parseInt(this.value, 10);
        }
    }

    isAce() {
        return this.value === "A";
    }

    getImagePath() {
        return `/cards/${this.toString()}.png`;
    }
}